src/hooks/registerHook.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License.
+
+import { AppStartHandler, AppTerminateHandler, PostInvocationHandler, PreInvocationHandler } from '@azure/functions';
+import * as coreTypes from '@azure/functions-core';
+import { Disposable } from '../utils/Disposable';
+import { tryGetCoreApiLazy } from '../utils/tryGetCoreApiLazy';
+import { AppStartContext } from './AppStartContext';
+import { AppTerminateContext } from './AppTerminateContext';
+import { PostInvocationContext } from './PostInvocationContext';
+import { PreInvocationContext } from './PreInvocationContext';
+
+function registerHook(hookName: string, callback: coreTypes.HookCallback): coreTypes.Disposable {
+    const coreApi = tryGetCoreApiLazy();
+    if (!coreApi) {
+        console.warn(
+            `WARNING: Skipping call to register ${hookName} hook because the "@azure/functions" package is in test mode.`

```

Comment:
Nit: should we extract into a simple function so we don't need to repeat the string and also in case we need to edit in the future? 

```
const warningMessage = (hookName: string) => `WARNING: Skipping call to register ${hookName} hook because the "@azure/functions" package is in test mode.`;
```

Reply:
Well the messages are different - one says "register" and one says "dispose". On my old team we would get our strings translated into other languages, and the translators preferred a little duplication over logic that dynamically constructed messages because the dynamic logic didn't work in other languages. We don't translate our strings, but I think I'm fine keeping "a little duplication" as a habit - at minimum it makes it easier to "ctrl+f" a string in the repo

Reply:
Oops, I should have read more carefully. I missed the slight change in wording. Yeah, in this case, there is no real benefit then.